spec/unit/knife/winrm_spec.rb

```diff
@@ -443,6 +443,35 @@
       expect { @winrm.run_with_pretty_exceptions }.to raise_error(SystemExit)
     end
 
+    context "impact of concurrency value" do
+      before(:each) do
+        allow(Chef::Knife::WinrmSession).to receive(:new).and_return(session)
+        @winrms = Chef::Knife::Winrm.new(['-C', '3', '-m', 'localhost somewhere knowhost', '-x', 'testuser', '-P', 'testpassword', '--winrm-authentication-protocol', 'basic', 'echo helloworld'])
+      end
+      context "when concurrency limit is set" do
```

Comment:
The shape of these is better, but I don't see that we're verifying the number of threads started in each case.

Reply:
@mp I have added the test case as below, But as per your point we cannot check impact of concurreny with number of threads running as it does the actually checking of host connection. which returns to fail due to dummy node. and mocking it will not give result of number of threads.